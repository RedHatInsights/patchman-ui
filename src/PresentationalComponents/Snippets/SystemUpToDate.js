import { Bullseye, EmptyState, EmptyStateBody, EmptyStateIcon, Title } from '@patternfly/react-core';
import { CheckCircleIcon } from '@patternfly/react-icons';
import React from 'react';
import { intl } from '../../Utilities/IntlProvider';
import messages from '../../Messages';

export const SystemUpToDate = () => (
    <Bullseye>
        <EmptyState style={{ paddingTop: 40 }}>
            <EmptyStateIcon icon={()=><CheckCircleIcon size='xl' color={'var(--pf-global--success-color--200)' }
                style={{ marginBottom: 15 }} />}/>
            <Title headingLevel="h5" size="lg">
                {intl.formatMessage(messages.statesNoApplicableAdvisories)}
            </Title>
            <EmptyStateBody>
                {intl.formatMessage(messages.statesSystemUpToDate)}
            </EmptyStateBody>
        </EmptyState>
    </Bullseye>
);
