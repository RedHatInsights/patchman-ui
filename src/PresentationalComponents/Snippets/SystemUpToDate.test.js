import { SystemUpToDate } from './SystemUpToDate';
import toJson from 'enzyme-to-json';

const wrapper = shallow(<SystemUpToDate />);

describe('SystemUpToDate', () => {
    it('Should match the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should render CheckCircleIcon', () => {
        const iconRender = wrapper.find('EmptyStateIcon').props().icon;
        const component = iconRender();
        expect(component.type.displayName).toEqual('CheckCircleIcon');
    });
});
