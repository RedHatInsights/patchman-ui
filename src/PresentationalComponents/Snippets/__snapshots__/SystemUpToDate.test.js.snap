// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemUpToDate Should match the snapshot 1`] = `
<Bullseye>
  <EmptyState
    style={
      Object {
        "paddingTop": 40,
      }
    }
  >
    <EmptyStateIcon
      icon={[Function]}
    />
    <Title
      headingLevel="h5"
      size="lg"
    >
      No applicable advisories
    </Title>
    <EmptyStateBody>
      This system is up to date, based on package information submitted at the most recent system check-in
    </EmptyStateBody>
  </EmptyState>
</Bullseye>
`;
